drivers/tapdisk-disktype.c

```diff
@@ -248,7 +248,7 @@ tapdisk_disktype_parse_params(const char *params, const char **_path)
 		return -ENAMETOOLONG;
 
 	memset(name, 0, sizeof(name));
-	safe_strncpy(name, params, len);
+	safe_strncpy(name, params, len + 1);
```

Comment:
Shouldn't `len` here denote the size of the destination, i.e. `sizeof(name)` instead of the size of the source `params`? (provided that the source IS null terminated)

Reply:
No. In this case we want to copy precisely `len` characters, a subset of the full `params` string.

It is perhaps a bit confusing using `safe_strncpy()`  in this way...
(Also, the memset is now unnecessary.)

Reply:
yes, I think in hindsight this is actually a case where it would be better to reinstate the original strncpy of the substring.